packages/builder/src/builderStore/store/automation/index.js

```diff
@@ -248,4 +248,39 @@ const automationActions = store => ({
     }
     await store.actions.save(newAutomation)
   },
+  replace: async (automationId, automation) => {
+    if (!automationId) {
+      return
+    }
+    if (!automation) {
+      store.update(state => {
+        // Remove the automation
+        state.automations = state.automations.filter(
+          x => x._id !== automationId
+        )
+        // Select a new automation if required
+        if (automationId === state.selectedAutomationId) {
+          store.actions.select(state.automations[0]?._id)
```

Comment:
NAB: won't do anything bad having this here I don't think, but is this needed? I would have expected that the URL state sync util would correctly recognise that the selected automation ID was no longer valid and trigger a redirect to select a new one.
That's what happens with other entities (screens etc).

Reply:
Yep, I did a bit a re-testing there and the state sync redirect stops it from becoming an issue  :thumbsup:

Reply:
Cool, lets just remove that then and we can merge after.